### PR TITLE
Strict mode

### PR DESCRIPTION
**Pull request by colin-msphere**
Launch Jenkins instances on strict mode cluster with auto-security mode detection.

### CODE REVIEW

File: testing/jenkins_remote_access.py [hunk 1]
```diff
@@ -171,6 +206,16 @@ def get_job_failures(service_name):
     return make_post(JENKINS_JOB_FAILURES, service_name)
 
 
+def change_mesos_creds(mesos_username, service_name):
```
Comment (kvish) [inline testing/jenkins_remote_access.py]: can we add a test for this method in test_scale_utils?

**Comment by benclarkwood** [comment]
This seems generally fine to me.

**Comment by benclarkwood** [comment]
Going to try it out now.

File: testing/sdk_dcos.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import sdk_cmd
+
+
+class DCOS_SECURITY(Enum):
+    disabled = 1
+    permissive = 2
+    strict = 3
+
+
+def get_metadata():
```
[the remark this answers is not in the document]
Comment (benclarkwood) [reply testing/sdk_dcos.py]: Also, this file should merge into upstream (dcos-commons) after this merges.

File: testing/jenkins_remote_access.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (colin-msphere) [reply testing/jenkins_remote_access.py]: Yes. The service account name (`jenkinsXYZ-principal`) that was created for this Jenkins instance is the username/principal that needs to be used with mesos. If theses do not match, the mesos driver has a bad time and the driver is in a `stop` state.

**Comment by colin-msphere** [comment]
@kvish - added test for changing mesos credentials.

**Pull request merged**